src/internal/_xdropRepeatsWith.js

```diff
@@ -0,0 +1,22 @@
+var _curry2 = require('./_curry2');
+var last = require('../last');
+
+
+module.exports = (function() {
+  function XDropRepeatsWith(pred, xf) {
+    this.xf = xf;
+    this.pred = pred;
+  }
+  XDropRepeatsWith.prototype['@@transducer/init'] = function() {
+    return this.xf.init();
```

Comment:
As are the `result` functions ... certainly opportunity to reduce redundancy.  I added [`transducer`](https://github.com/transduce/transduce#transducerstep-result-init) to alleviate it in my `transduce` project, but it's not very ramda-y with the optional args...